cypress/integration/articleFlows/mangageFlows/lockDiscussion.spec.js

```diff
@@ -0,0 +1,107 @@
+describe('Lock discussion', () => {
+  const getDiscussionLockButton = () =>
+    cy.findByRole('link', {
+      name: /Lock discussion/,
+    });
+
+  const getDiscussionUnlockButton = () =>
+    cy.findByRole('link', {
+      name: /Unlock discussion/,
+    });
+
+  const getDiscussionLockSubmitButton = () => cy.get('.crayons-btn--danger');
+
+  const exampleReason = 'Discussion lock example reason!';
+  const exampleNotes = 'Discussion lock example notes!';
+
+  describe('locking the discussion', () => {
+    beforeEach(() => {
+      cy.testSetup();
+      cy.fixture('users/articleEditorV1User.json').as('user');
+
+      cy.get('@user').then((user) => {
+        cy.loginUser(user).then(() => {
+          cy.createArticle({
+            title: 'Test Article',
+            tags: ['beginner', 'ruby', 'go'],
+            content: `This is a test article's contents.`,
+            published: true,
+          }).then((response) => {
+            cy.visit(`${response.body.current_state_path}/manage`);
+          });
+        });
+      });
+    });
+
+    it('should allow a user to lock a discussion', () => {
+      getDiscussionLockButton().click();
+      getDiscussionLockSubmitButton().click();
+
+      getDiscussionUnlockButton().should('exist');
+    });
+
+    it('should ask the user to confirm the discussion lock', () => {
+      getDiscussionLockButton().click();
+
+      cy.findByRole('heading', {
+        name: 'Are you sure you want to lock the discussion on this article?',
+      }).should('exist');
+
+      cy.findByRole('button', {
+        name: /Lock discussion/,
+      }).should('exist');
+    });
+
+    it('should allow a user to supply a reason and notes', () => {
+      getDiscussionLockButton().click();
+      cy.get('input[name="discussion_lock[notes]"]').should('exist');
+      cy.get('input[name="discussion_lock[reason]"]').should('exist');
```

Comment:
```suggestion
cy.findByRole('textbox', {
        name: 'Reason for locking the discussion (optional) - this will be publicly displayed',
      }).should('exist');
      cy.findByRole('textbox', {
        name: 'Notes (optional) - this is only visible to you and admins',
      }).should('exist');
```

I think we should change this, so it's more focused on the user-facing label

Reply:
Similar to - https://github.com/forem/forem/pull/13905#discussion_r647787780, I originally didn't think the text mattered as much.

Regardless, I implemented your suggestions.

Reply:
I think the text matters - e2e tests are really to test from a user's point of view. If we accidentally push a change that unlinks the label from the input, these tests will catch that if we use the input name